### ISSUE TITLE
snt.Conv2DTranspose and tf.layers.conv2d_transpose yield different results

### ISSUE THREAD

**Issue by ferreirafabio** [deepmind/sonnet#110]
The sonnet documentation states the about the Conv2DTranspose:

> This acts as a light wrapper around the TensorFlow op `tf.nn.conv2d_transpose`
> abstracting away variable creation and sharing.

```tf.layers.conv2d_transpose(inputs, filters=64, kernel_size=1, strides=5, padding='valid')```
--> yields shape `(?, 5, 5, 64)`

`snt.Conv2DTranspose(64, 1, 5, padding='VALID')(inputs)`
--> yields shape `(?, 1, 1, 64)`

for `inputs` shape `(?, 1, 1, 218)`

Changing the semantics of the called parameters from `input_channels` (tensorflow) to `output_channel` (Sonnet) in the API to me does not state not a 'light' wrapper and either the wording or the code should therefore be adjusted in Sonnet.

**Comment by malcolmreynolds** [comment]
Hi @ferreirafabio 

 I'm not exactly sure what you are asking - do you think that the argument orders of `snt.Conv2DTranspose` and `tf.layers.conv2d_transpose` should necessarily be the same / in the same order? This is not something we try and do at all, and we can't make the order of arguments the same as the underlying tf.nn.conv2d_transpose op because the first two arguments of that function are the tensor to operate on and the tensor containing the filters. We have chosen different names for some arguments, generally for consistency within the Sonnet library.

You can get your desired result if you are explicit with your kwargs, ie:

```
inputs = tf.placeholder(name='foo', shape=[None, 1, 1, 218],dtype=tf.float32)
snt.Conv2DTranspose(output_channels=64, kernel_shape=1, stride=5, padding='VALID')(inputs)
```

This yields shape (?, 5, 5, 64) output shape.

Please let me know if I'm misunderstanding?

**Comment by tomhennigan** [comment]
Closing out some old issues, please reopen if this is still a blocker.
